From: Paul Srisoontorn <psrisoontorn@nvidia.com>
Date: Mon, 12 Jan 2026 00:00:00 +0000
Subject: [PATCH] drivers: ipm: use DRAM for IPC channel addresses

Change IPC channel addresses from SRAM to DRAM for inter-processor
communication. This provides larger buffers (1MB) for SSP and BootMCU
communication channels.

Upstream-Status: Pending

Signed-off-by: Paul Srisoontorn <psrisoontorn@nvidia.com>
---
 include/zephyr/drivers/ipm.h | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/include/zephyr/drivers/ipm.h b/include/zephyr/drivers/ipm.h
index abcdef0..1234567 100644
--- a/include/zephyr/drivers/ipm.h
+++ b/include/zephyr/drivers/ipm.h
@@ -30,7 +30,7 @@ extern "C" {
 #endif
 
 #define IPC_SHARE_MEM_SRAM_SIZE		(0x1000) /* 4KB */
-
+
 #define IPC_SHARE_MEM_LOCAL_S_CA35_DATA                                                            \
 	__attribute__((__section__("IPC_SHARE_MEM_LOCAL_S_CA35_DATA")))
 #define IPC_SHARE_MEM_LOCAL_NS_CA35_DATA                                                           \
@@ -44,17 +44,18 @@ extern "C" {
 #define IPC_SHARE_MEM_LOCAL_BOOTMCU_CALIPTRA                                                       \
 	__attribute__((__section__("IPC_SHARE_MEM_LOCAL_BOOTMCU_CALIPTRA")))
 
-#define IPC_CHANNEL_ID_CH0			0
-#define IPC_CHANNEL_0_SSP_IN_ADDR		(0x70004000)
-#define IPC_CHANNEL_0_SSP_OUT_ADDR		(0x70002000)
-#define IPC_CHANNEL_0_BOOTMCU_IN_ADDR		(0x70004000)
-#define IPC_CHANNEL_0_BOOTMCU_OUT_ADDR		(0x70002000)
-#define IPC_CHANNEL_ID_CPTRA			1
-#define IPC_CHANNEL_1_SSP_IN_ADDR		(0x70005000)
-#define IPC_CHANNEL_1_SSP_OUT_ADDR		(0x70003000)
-#define IPC_CHANNEL_1_BOOTMCU_IN_ADDR		(0x70005000)
-#define IPC_CHANNEL_1_BOOTMCU_OUT_ADDR		(0x70003000)
-#define IPC_CHANNEL_1_BUFF_SIZE			(1024 * 1024)
+// Use IPC in DRAM instead of SRAM
+#define IPC_CHANNEL_ID_CH0             0
+#define IPC_CHANNEL_0_SSP_IN_ADDR      (0x1000000 + (0 * 1024 * 1024))
+#define IPC_CHANNEL_0_SSP_OUT_ADDR     (0x1000000 + (1 * 1024 * 1024))
+#define IPC_CHANNEL_0_BOOTMCU_IN_ADDR  (0xAD000000 + (0 * 1024 * 1024))
+#define IPC_CHANNEL_0_BOOTMCU_OUT_ADDR (0xAD000000 + (1 * 1024 * 1024))
+#define IPC_CHANNEL_ID_CPTRA           1
+#define IPC_CHANNEL_1_SSP_IN_ADDR      (0x1000000 + (2 * 1024 * 1024))
+#define IPC_CHANNEL_1_SSP_OUT_ADDR     (0x1000000 + (3 * 1024 * 1024))
+#define IPC_CHANNEL_1_BOOTMCU_IN_ADDR  (0xAD000000 + (2 * 1024 * 1024))
+#define IPC_CHANNEL_1_BOOTMCU_OUT_ADDR (0xAD000000 + (3 * 1024 * 1024))
+#define IPC_CHANNEL_1_BUFF_SIZE        (1024 * 1024)
 
 /**
  * @typedef ipm_callback_t
-- 
2.25.1
